pkg/gateway/operations/deleteobject.go

```diff
@@ -31,12 +31,12 @@ func (controller *DeleteObject) HandleAbortMultipartUpload(w http.ResponseWriter
 	mpu, err := o.MultipartTracker.Get(ctx, uploadID)
 	if err != nil {
 		o.Log(req).WithError(err).Error("upload id not found in tracker")
-		_ = o.EncodeError(w, req, err, gatewayerrors.Codes.ToAPIErr(gatewayerrors.ErrNoSuchKey))
+		_ = o.EncodeError(w, req, err, gatewayerrors.Codes.ToAPIErr(gatewayerrors.ErrNoSuchKey, nil))
```

Comment:
Any reason not to pass the actual error instead of nil to all of these calls?

Reply:
We decided to limit the scope of the current fix to the CopyObject API.
We need to consider a more holistic approach for the actual solution that might require a deeper fix